### PR TITLE
Fix scale change detection at runtime on wayland.

### PR DESCRIPTION
**Pull request by tarmack**
Without this any changes to the output scale after kitty is started is never applied.

### CODE REVIEW

File: glfw/wl_window.c [hunk 1]
```diff
@@ -1041,6 +1048,7 @@ void _glfwPlatformSetWindowSize(_GLFWwindow* window, int width, int height)
 {
     window->wl.width = width;
     window->wl.height = height;
+    checkScaleChange(window);
```
Comment (kovidgoyal) [inline glfw/wl_window.c]: this should go into the handleConfigure functions ratherthan inside the set window size function. I believe other parts of glfw can call this function as well.

Comment (tarmack) [reply glfw/wl_window.c]: You are absolutely right, it was not working at all like this before (application would hang). Substituting `_glfwInputWindowContentScale` for the direct resize fixed that, I think this solution is much cleaner in general.

Comment (kovidgoyal) [reply glfw/wl_window.c]: But doesn't this mean that if there is a scale change but not a size change resizeWindow will not be called? I think this entire thing needs to be re-factored. The size and scale should all be checked independently in configure callbacks. And then the appropriate input window callback should be called. And if either size or scale has changed, the resizeWindow function should be called.

Comment (tarmack) [reply glfw/wl_window.c]: You are right, this thing needs a refactor really badly. There is actually a ton of duplication that needs removing as well. I'll get back when I have something.

**Comment by kovidgoyal** [comment]
This should be fixed in current master.

**Pull request closed without merging**